changelog/fix-6782-support-phone-dev-mode

```diff
@@ -0,0 +1,4 @@
+Significance: patch
+Type: fix
+
+Allow test support phone in dev mode
```

Comment:
May update the title of the PR too.

Reply:
Not sure what to change here? Allow "test phone number" (as Stripe calls it) 
<img width="376" alt="image" src="https://github.com/Automattic/woocommerce-payments/assets/32092402/5828ebb7-b29d-410d-9de7-0c7ca8dd1db7">
 for support phone number in "dev mode"
I've tried to  modify it to make it more clear.

Reply:
🤦 @jessy-p - sorry, I meant the title of this PR, not the changelog. 
Currently, it's `Skip support phone validation in dev mode`. You would update it to `Allow test support phone in dev mode`.